packages/esm-patient-registration-app/src/patient-registration/field/id/id-field.component.tsx

```diff
@@ -3,48 +3,113 @@ import { IdentifierInput } from '../../input/custom-input/identifier/identifier-
 import styles from '../field.scss';
 import { useTranslation } from 'react-i18next';
 import { PatientRegistrationContext } from '../../patient-registration-context';
-import { Button } from 'carbon-components-react';
+import { Button, SkeletonText } from 'carbon-components-react';
 import { ArrowRight16 } from '@carbon/icons-react';
 import { useLayoutType, useConfig } from '@openmrs/esm-framework';
-import { PatientIdentifierValue } from '../../patient-registration-types';
+import {
+  FormValues,
+  IdentifierSource,
+  PatientIdentifierType,
+  PatientIdentifierValue,
+} from '../../patient-registration-types';
 import IdentifierSelectionOverlay from './identifier-selection-overlay';
-import { FieldArray } from 'formik';
 import { ResourcesContext } from '../../../offline.resources';
 
-export const IdField: React.FC = () => {
+export function setIdentifierSource(
+  identifierSource: IdentifierSource,
+  identifierValue: string,
+  initialValue: string,
+): {
+  identifierValue: string;
+  autoGeneration: boolean;
+  selectedSource: IdentifierSource;
+} {
+  const autoGeneration = identifierSource?.autoGenerationOption?.automaticGenerationEnabled;
+  return {
+    selectedSource: identifierSource,
+    autoGeneration,
+    identifierValue: autoGeneration
+      ? 'auto-generated'
+      : identifierValue !== 'auto-generated'
```

Comment:
This is saying that if the identifier value is set to 'auto-generated' we would change it to initialValue, otherwise we set it to identifierValue ?? Are you sure this logic is correct?

Reply:
Yes, if source was initially autoGeneration enabled, then the value will be 'auto-generated', but if we switch to a manual source, then we have to change back the 'auto-generated' to the intialValue, which will either be an empty string, or initial identifier value.